lib/widgets/features/profile/edit_profile_preferred_language.dart

```diff
@@ -36,31 +53,32 @@ class _EditPreferredLanguageScreenState
 
   @override
   Widget build(BuildContext context) {
+    if (!pageRoute.isCurrent) return const SizedBox.shrink();
     final AppLocalizations l10n = AppLocalizations.of(context)!;
-    final ThemeData theme = Theme.of(context);
 
     return EditTemplate(
-      title: l10n.profileEditLanguagePreferred,
-      body: Padding(
-        padding: const EdgeInsets.all(Insets.paddingMedium),
-        child: Column(
-          children: [
-            Padding(
-              padding: const EdgeInsets.all(Insets.paddingSmall),
-              child: Text(
-                l10n.profileEditLanguagePreferredSubtitle,
-                style: theme.textTheme.bodyMedium!
-                    .copyWith(color: theme.colorScheme.secondary),
-                textAlign: TextAlign.center,
-              ),
-            ),
-            AutocompletePicker(
-              controller: _preferredLanguagesController,
-              options: _contentProvider.languageOptions!
-                  .map((e) => e.translatedValue!)
-                  .toList(),
-            ),
-          ],
+      title: l10n.profileEditSectionAboutLanguagePreferredTitle,
+      subtitle: l10n.profileEditSectionAboutLanguagePreferredSubtitle,
+      scaffoldBuilder: buildPageRouteScaffold,
+      body: AutocompletePicker(
+        singleSelect: true,
+        controller: _preferredLanguagesController,
+        options: _contentProvider.languageOptions!
+            .map((e) => e.translatedValue!)
+            .toList(),
+        selectedOptions: _initialValue,
+      ),
+      editUserProfile: () => _userProvider.updateUserData(
+        input: Input$UserInput(
+          id: widget.userData.id,
+          preferredLanguageTextId: _preferredLanguagesController.getTags
+              ?.map(
+                (t) => _contentProvider.languageOptions!
+                    .firstWhere((o) => o.translatedValue! == t)
```

Comment:
probably better to compare the `textId`s rather than the `translatedValue`s. in theory the translated value should work fine, but i don't know if it will work with every language (e.g. does comparing two arabic strings with `==` always work in Dart? no idea)